internal/provider/provider.go

```diff
@@ -315,6 +325,7 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 			DisableTerraformPartnerID:   d.Get("disable_terraform_partner_id").(bool),
 			Features:                    expandFeatures(d.Get("features").([]interface{})),
 			StorageUseAzureAD:           d.Get("storage_use_azuread").(bool),
+			UseMSAL:                     d.Get("use_msal").(bool),
```

Comment:
```suggestion
			UseMSAL:                     func() bool {
				if features.ThreePointOh() {
					return true
				}
				
				return d.Get("use_msal").(bool)
			}(),
```

Reply:
As above, might be better to consolidate this as a DefaultFunc for the `use_msal` property?

Reply:
Sure that'd work 👍